src/transformers/models/seggpt/modeling_seggpt.py

```diff
@@ -817,8 +817,11 @@ def forward(
         # and reconstructed together (In-Context Painting).
         if bool_masked_pos is None:
             num_patches = self.embeddings.patch_embeddings.num_patches
-            bool_masked_pos = torch.zeros(num_patches, dtype=torch.bool).to(pixel_values.device)
-            bool_masked_pos[num_patches // 2 :] = 1
+            bool_masked_pos_zeros = torch.zeros(num_patches // 2, dtype=torch.bool, device=pixel_values.device)
+            bool_masked_pos_ones = torch.ones(
+                num_patches - num_patches // 2, dtype=torch.bool, device=pixel_values.device
```

Comment:
is this because of the same reason as torch.export not supporting in-place operation?

Reply:
supports, but not always, in some cases it fails, I didn't go deep in details, they recommend make it not in-place to fix